react-components/styling-components.md

```diff
@@ -0,0 +1,243 @@
+
+Following these guidelines, to style her `StudentList` component, she'll create a file `src/components/StudentList.css`.
+
+### !callout-info
+
+## CSS File Recommendations
+
+Like other parts of React, the name and location of this file isn't mandatory, but a useful recommended pattern.
+
+### !end-callout
+
+### Populate the CSS File
+
+At this moment, Sofia can populate `src/components/StudentList.css` with the following style. This style will apply a bright red box to every element, and it's an effective way to check that CSS is working.
+
+```css
+* {
+  border: solid 2px red;
+}
+```
+
+### Import CSS File
+
+To include the new CSS file, Sofia should import it into the `StudentList` component file.
+
+```js
+import "./StudentList.css";
+```
+
+While Sofia is in this file, she'll add some student content to make this app more interesting.
+
+In order to return only _one_ JSX object, Sofia needs to wrap her content into a new outer element.
+
+```js
+import React from "react";
+import "./StudentList.css";
+
+const StudentList = () => {
+  return (
+    <section>
+      <h2>Student List</h2>
+      <ul>
+        <li>Student A</li>
+        <li>Student B</li>
+        <li>Student C</li>
+      </ul>
+    </section>
+  );
+};
+
+export default StudentList;
+```
+
+We see our CSS is applied, just by importing the CSS file!
+
+![](../assets/react-components_styling-components_red-boxes.png)
+
+### Using `class`/`className`
+
+Sofia should replace her CSS with some reasonable styles.
+
+We can use any valid CSS selectors, such as element selectors, classes, or IDs.
+
+```css
+h2 {
+  color: darkblue;
+  font-style: oblique;
+}
+```
+
+![](../assets/react-components_styling-components_element-selector.png)
+
+However, there is one exception: in order to set the `class` attribute to any element in JSX, we must use the attribute `className`.
+
+Sofia can modify her `StudentList` component to include some classes with `className`:
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = () => {
+    return (
+        <section>
+            <h2 className="student-list__heading">Student List</h2>
+            <ul className="student-list">
+                <li>Student A</li>
+                <li>Student B</li>
+                <li>Student C</li>
+            </ul>
+        </section>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+And create CSS rule-sets that select those classes:
+
+```css
+.student-list__heading {
+  color: darkblue;
+  font-style: oblique;
+}
+
+.student-list {
+  list-style-type: square;
+}
+```
+
+We can even check our browser Dev Tools and inspect the rendered HTML. The `className` attribute defined in JSX has turned into the `class` attribute in rendered HTML.
+
+![](../assets/react-components_styling-components_classname-render.png)
+
+## More CSS
+
+We can apply our current React and CSS knowledge to our projects.
+
+<!-- I'm emphasizing this in order to pave the way for conditional logic -->
+
+### Embedding Classes
```

Comment:
to me, injecting sound like a better description of what we're doing
adding a bit more detail about what we meant by the similar component situation might be helpful
```suggestion
### Injecting Classes

We were able to store component objects in variables, then use `{varName}` syntax in our JSX to inject that component into the JSX structure.

We can do the same thing with CSS classes.
```

Reply:
I like the word inject and the description!

I'm unsure if you're suggesting "inject" over "embed" for this specific situation, or to prefer "inject" in general.

As a heads up, I think I end up using the word "embed" specifically a lot in the `props` PR. I pulled this term from [React's docs on JSX](https://reactjs.org/docs/introducing-jsx.html#embedding-expressions-in-jsx)! Just giving context for why you'll see it a bunch.